build_tools/utils.py

```diff
@@ -294,24 +295,32 @@ def copy_common_headers(
         shutil.copy(path, new_path)
 
 
+def pip_or_uv() -> List[str]:
+    if find_spec("pip") is not None:
+        return [sys.executable, "-m", "pip"]
+    else:
+        return ["/usr/bin/env", "uv", "pip"]
+
+
 def install_and_import(package):
     """Install a package via pip (if not already installed) and import into globals."""
     main_package = package.split("[")[0]
-    subprocess.check_call([sys.executable, "-m", "pip", "install", package])
+    subprocess.check_call([*pip_or_uv(), "install", package])
     globals()[main_package] = importlib.import_module(main_package)
 
 
 def uninstall_te_wheel_packages():
-    subprocess.check_call(
-        [
-            sys.executable,
-            "-m",
-            "pip",
-            "uninstall",
-            "-y",
-            "transformer_engine_cu12",
-            "transformer_engine_torch",
-            "transformer_engine_paddle",
-            "transformer_engine_jax",
-        ]
-    )
+    if find_spec("pip") is not None:
+        subprocess.check_call(
+            [
+                sys.executable,
+                "-m",
+                "pip",
+                "uninstall",
+                "-y",
+                "transformer_engine_cu12",
+                "transformer_engine_torch",
+                "transformer_engine_paddle",
+                "transformer_engine_jax",
+            ]
+        )
```

Comment:
This is a reasonable change that I would accept on its own. It doesn't fully solve the root issue (the package structure is different when installing from PyPI or from source), so I could imagine users running into problems if they install TE multiple times. But it seems fair to expect users to handle this on their own if they are using a non-standard build environment.